### PR TITLE
feat: sort top_k and return top_k Document from DocumentArray

### LINKED ISSUE #3467: DocumentArray `sort` should also get `top_k` to allow for more efficient sorting
**Describe the feature**
Right now `sort` is done on all the `Document` collection, I would like to sort only up to a `top_k`.

### PR DESCRIPTION
**Pull request by sauravgarg540**
This PR resolves #3467

## Resolution:
- Sort top_k and return top_k `Document` from `DocumentArray`
- added unit test

### CODE REVIEW

File: jina/types/arrays/document.py [hunk 1]
```diff
@@ -375,14 +375,19 @@ def reverse(self):
...
         """
         Sort the items of the :class:`DocumentArray` in place.
 
         :param key: key callable to sort based upon
+        :param top_k: sort only upto top_k Document from DocumentArray
         :param args: variable set of arguments to pass to the sorting underlying function
         :param kwargs: keyword arguments to pass to the sorting underlying function
         """
+        # Only for DocumentArray
+        if isinstance(self._pb_body, list):
```
Comment (JoanFM) [inline jina/types/arrays/document.py:388]: Maybe it was not clear from the documentation. The point is to only sort the `first top_k` but of the sorted list.

There are sorting algorithms that do not sort a complete collection but that only guarantee that they first `top_k` elements are sorted while the rest may not be

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by hanxiao** [comment]
i dont think this implementation is needed , how is it different from `.sort()[:k]`

closing

**Comment by JoanFM** [comment]
> i dont think this implementation is needed , how is it different from `.sort()[:k]`
> 
> closing

agree, this implementation is not the proper one. This is why there are changes requested

**Comment by sauravgarg540** [comment]
Hey @JoanFM, thank you for the clarification on the issue description. I was already in doubt that why only sort first K values in a collection. I will create another PR with the correct implementation.

**Pull request closed without merging**